File: lingting-grpc/src/main/java/live/lingting/framework/grpc/exception/GrpcExceptionHandler.java
package live.lingting.framework.grpc.exception;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author lingting 2024-03-27 09:43
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface GrpcExceptionHandler {

	Class<? extends Throwable>[] value() default {};

}

File: lingting-grpc/src/main/kotlin/live/lingting/framework/grpc/exception/GrpcExceptionHandler.kt
package live.lingting.framework.grpc.exception

import kotlin.reflect.KClass

/**
 * @author lingting 2024-03-27 09:43
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class GrpcExceptionHandler(vararg val value: KClass<out Throwable> = [])
